.github/workflows/ci.yml

```diff
@@ -46,5 +46,11 @@ jobs:
       - name: vet
         run: go vet ./...
 
+      - name: ensure compilation
+        env:
+          GOOS: linux
+        run: go build
+
       - name: gotestsum
+        if: ${{ platform == 'ubuntu-latest' }}
```

Comment:
I should add a comment here on why we are doing this, it is because many of the containers we use in tests will not run on anything besides Linux.